### PR TITLE
#18187 Using whitespace analyzer with the host aliases

### PR DESCRIPTION
**Pull request by freddyucv**
Using the whitespace analizer to the host aliases to avoid the tokenizes

https://github.com/dotCMS/core/compare/issue-18187-Aliases-not-working-correctly?expand=1#diff-44746dc0a1c617ec2edbabe9d6fedf39R72

https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-whitespace-analyzer.html

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java [hunk 1]
```diff
@@ -14,22 +14,38 @@
     private final long currentTime = System.currentTimeMillis();
 
     private String name = "test" + currentTime + ".dotcms.com";
+    private String aliases = null;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Avoid using redundant field initializer for 'aliases'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java [hunk 2]
```diff
@@ -14,22 +14,38 @@
...
+    private boolean isDefault = false;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:18]: Codacy found an issue: [Avoid using redundant field initializer for 'isDefault'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Field aliases has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java [hunk 3]
```diff
@@ -639,4 +639,25 @@ private void addPermission(final Role role, final Host host)
...
+    /**
+     * Method to test: {@link HostAPI#findByAlias(String, User, boolean)}
+     * When create two host: first one with alias equals to demo.dotcms.com and second  one with alias equals to not-demo.dotcms.com
+     *      and find by  demo.dotcms.com
+     * Should return the first one
+     */
+    @Test
+    public void shouldReturnHostByAlias() throws DotSecurityException, DotDataException {
+        final Host host = new SiteDataGen().aliases("demo.dotcms.com").nextPersisted();
+        final Host host_2 = new SiteDataGen().aliases("not-demo.dotcms.com").setDefault(true).nextPersisted();
```
Comment (wezell) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java:651]: Please write a test that tests where there are multiple host aliases separated by \r and \n and also a tab and space. 

Also, the broken case is where there are multiple hosts that start with `not-` e.g. `prod-client.dotcms.com` and `prod-anotherclient.dotcms.com`.  Please write a test for this as well - as it is the broken case.

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Avoid using redundant field initializer for 'aliases'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:18]: Codacy found an issue: [Avoid using redundant field initializer for 'isDefault'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Field aliases has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java:651]: done

https://github.com/dotCMS/core/pull/18829/commits/1c06c59de80cc2b15782eceb455d7b8d27d149ec

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java [hunk 4]
```diff
@@ -14,22 +14,38 @@
     private final long currentTime = System.currentTimeMillis();
 
     private String name = "test" + currentTime + ".dotcms.com";
+    private String aliases;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Field aliases has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Field aliases has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Field aliases has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java [hunk 5]
```diff
@@ -39,6 +39,8 @@
 import java.util.Optional;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
+
+import io.vavr.API;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java:43]: Codacy found an issue: [Avoid unused imports such as 'io.vavr.API'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/SiteDataGen.java:17]: Codacy found an issue: [Field aliases has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5809706)

**Review by jdotcms — APPROVED** [review]

**Review by wezell — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**